scripts/generating_yaml.py

```diff
@@ -0,0 +1,286 @@
+
+    def lens_sn_widget(lens_serial_number):
+        """Function used to create Jupyter widget.
+        It takes the parameter chosen from the widget and returns it such that it can be used as a variable.
+
+        Args:
+            lens_serial_number (str) : the serial number of the lens device as selected from the widget.
+
+        Returns:
+            The result of the widget; the chosen lens serial number
+
+        """
+        return lens_serial_number
+
+    def filter_ID_widget(filter_ID_code):
+        """Function used to create Jupyter widget.
+        It takes the parameter chosen from the widget and returns it such that it can be used as a variable.
+
+        Args:
+            filter_ID_code (str) : the ID number of the lens as selected from the widget.
+
+        Returns:
+            The result of the widget; the chosen filter ID number
+
+        """
+        return filter_ID_code
+
+    def serial_to_usb_widget(serial_into_USBhub_port):
+        """Function used to create Jupyter widget.
+        It takes the parameter chosen from the widget and returns it such that it can be used as a variable.
+
+        Args:
+            serial_into_USBhub_port (str) : the port number of the USB Hub that the Serial Adaptor is plugged into as selected from the widget.
+
+        Returns:
+            The result of the widget; the chosen USB port number
+
+        """
+        return serial_into_USBhub_port
+
+    def camera_to_serial_widget(camera_into_serial_port):
+        """Function used to create Jupyter widget.
+        It takes the parameter chosen from the widget and returns it such that it can be used as a variable.
+
+        Args:
+            camera_into_serial_port (str) : the port number of the Serial Adaptor that the camera is plugged into as selected from the widget.
+
+        Returns:
+            The result of the widget; the chosen serial port number
+
+        """
+        return camera_into_serial_port
+
+    def usbhub_sn_widget(USBhub_SN):
+        """Function used to create Jupyter widget.
+        It takes the parameter chosen from the widget and returns it such that it can be used as a variable.
+
+        Args:
+            USBhub_SN (str) : the serial number of the USB Hub as selected from the widget.
+
+        Returns:
+            The result of the widget; the chosen USB Hub serial number
+
+        """
+        return USBhub_SN
+
+    def camera_to_usb_widget(camera_into_USBhub_port):
+        """Function used to create Jupyter widget.
+        It takes the parameter chosen from the widget and returns it such that it can be used as a variable.
+
+        Args:
+            camera_into_USBhub_port (str) : the port number of the USB Hub that the camera is plugged into as selected from the widget.
+
+        Returns:
+            The result of the widget; the chosen USB port number
+
+        """
+        return camera_into_USBhub_port
+
+    birger_sn = data['birger_SN']
```

Comment:
Maybe provide a 1-line comment before this block briefly stating what it is about to do.